source/base/parameter_handler.cc

```diff
@@ -1219,18 +1219,28 @@ ParameterHandler::set(const std::string &entry_string, const bool new_value)
 
 std::ostream &
 ParameterHandler::print_parameters(std::ostream &    out,
-                                   const OutputStyle style,
-                                   const bool        sort_alphabetical) const
+                                   const OutputStyle style) const
 {
   AssertThrow(out, ExcIO());
 
+  Assert(
+    (((style & XML) > 0) + ((style & JSON) > 0) + ((style & PRM) > 0) +
+     ((style & Description) > 0) + ((style & LaTeX) > 0)),
+    ExcMessage(
+      "You have chosen either no or multiple style formats. You can choose between: PRM, Description, LaTeX, XML, JSON."));
+
+  Assert(((style & LaTeX) && (style & Short)) == false,
+         ExcMessage("It is not possible to request short LaTex."));
+  Assert(((style & Description) && (style & Short)) == false,
+         ExcMessage("It is not possible to request short Description."));
```

Comment:
Personally, I would prefer to have all these asserts in one private member function that ensures consistency of the output style, as a means to satisfy our design goals from #9888 and to highlight that this check is related to the data structure OutputStyle and not to a specific function? If I understood it correctly, more functions might need these checks, so a copy-paste of this code is forseeable otherwise.

Reply:
That’s actually a very nice suggestion. @peterrum, could you wrap up the checks in a function returning a Boolean, and assert on the returned value?

Reply:
I went with @bangerth' suggestion with the free function in an anonymous namespace.